warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/EventMapperTest.java

```diff
@@ -212,6 +213,39 @@ public void errorEventWithZeroTimestampNotDropped() throws IOException, Interrup
         assertEquals(4, written.size());
     }
 
+    @Test
+    public void testDiscardInterval() throws IOException, InterruptedException {
+        // event date < now minus discard interval
+        long tomorrow = -1L * UIDConstants.MILLISECONDS_PER_DAY;
+        conf.setLong(record.getDataType().typeName() + "." + EventMapper.DISCARD_INTERVAL, tomorrow);
+
+        eventMapper.setup(mapContext);
+        eventMapper.map(new LongWritable(1), record, mapContext);
+        eventMapper.cleanup(mapContext);
+
+        Multimap<BulkIngestKey,Value> written = TestContextWriter.getWritten();
+
+        // discard interval lower bound is tomorrow. data from today should be dropped
+        assertEquals(0, written.size());
+    }
+
+    @Test
+    public void testFutureDiscardInterval() throws IOException, InterruptedException {
+        // event date > now plus future discard interval
+        long yesterday = -1L * UIDConstants.MILLISECONDS_PER_DAY;
+        conf.setLong(record.getDataType().typeName() + "." + EventMapper.DISCARD_INTERVAL, yesterday);
+
+        eventMapper.setup(mapContext);
+        eventMapper.map(new LongWritable(1), record, mapContext);
+        eventMapper.cleanup(mapContext);
+
+        Multimap<BulkIngestKey,Value> written = TestContextWriter.getWritten();
+
+        // future discard is yesterday. data from today should be dropped
+        assertEquals(0, written.size());
+
+    }
```

Comment:
I recommend having at least two tests that set the future discard interval to a non-zero value:
- one that eliminates nothing
- one that eliminates everything

Reply:
agreed.

Reply:
Done, see https://github.com/NationalSecurityAgency/datawave/pull/2271